llvm/test/CodeGen/Hexagon/fcmp-nan.ll

```diff
@@ -0,0 +1,189 @@
+; RUN: llc -march=hexagon < %s | FileCheck %s
```

Comment:
Do we need to consider running tests like this in other floating point modes like HVX floating point (formerly qfloat) and/or HVX IEEE fp?

Reply:
That's a good point. I'll check what the behavior is for HVX fp/ieee and if there are tests to check this.

Reply:
This nan check is not needed for vector fp comparisons. The `vcmp.gt` instruction returns false if any of the values are nan's. There's already a lit test to check this here: https://github.com/llvm/llvm-project/blob/main/llvm/test/CodeGen/Hexagon/autohvx/vector-compare-float.ll